### DISCUSSION TITLE
Is "yarpmotorgui" an appropriate name for a joint interface?

### DISCUSSION THREAD

**Discussion by DanielePucci** [robotology/community#97] [Q&A] [answered]
Hi,
I was wondering if `yarpmotorgui` is the right name for an interface that is, at all effects and purposes, a joint interface.

I propose, `yarpjointgui` :)

@pattacini @randaz81 @vtikha

**Comment by DanielePucci** [comment]
@iron76 @lornat75

**Comment by traversaro** [comment]
I think no one in history was confused by this: joint are controlled by motors, and if you know the difference between joint-level or motor-level control you already know so much about the yarp/icub infrastructure not to be confused by the name of the `yarpmotorgui`. 

I don't think that the trouble of renaming (and educating all the users about the new name/modifyng scripts and tutorials mentioning the old name) is worthy.

**Comment by pattacini** [comment]
I'm afraid @traversaro is right.

**Comment by vtikha** [comment]
@DanielePucci you have a valid point, but @traversaro's answer could not have been any clearer.

**Comment by francesco-romano** [comment]
``` bash
if [ computer == Pucci's computer ]; then
  ln -s yarpmotorgui yarpjointgui
fi
```

**Comment by naveenoid** [comment]
@francesco-romano :joy:

**Comment by DanielePucci** [comment]
Bah, the name is semantically wrong, especially in light of coupled joints where motors and joints cannot be linked by a one-to-one relation during motion. But, it's up to you. It was just a proposal.

**Comment by pattacini** [comment]
I'm afraid @DanielePucci is right.

**Comment by pattacini** [comment]
(but we can keep the coupling secret :smirk:)

**Comment by traversaro** [comment, marked as answer]
> Bah, the name is semantically wrong, especially in light of coupled joints where motors and joints cannot be linked by a one-to-one relation during motion. But, it's up to you. It was just a proposal. 

Let me amend my reply: 

> I think no one in history was confused by this: joint are controlled by motors, and if you know the difference between joint-level and motor-level control or **about coupled joints**, you already know so much about the yarp/icub infrastructure not to be confused by the name of the yarpmotorgui. 

More seriously, if we want to avoid confusion about joint/motor, I think improving documentation on coupling support in yarp ControlBoard would be a much more useful use of our time then changing the name and then manually update outdated documentation/code or educate users about the new name. 

Just to give a (clearly biased) example: I still type `robotMotorGui` most of the times I want to use `yarpmotorgui`, and I would literarly _hate_ any upstream project that changes the name of one of its most used tools for two times in less then a year.

**Comment by barbalberto** [comment]
:+1:  x Silvio :-)

**Comment by DanielePucci** [comment]
Being conservative is the (other) root of all evil.

 With that said, I understand your point.

**Comment by pattacini** [comment]
@DanielePucci join the dark side.

**Comment by DanielePucci** [comment]
Unfortunately, I did :D

**Comment by drdanz** [comment]
There was a discussion about the new name when we moved it in yarp, one of the proposed names was yarpjointmover. I still hate all the options, especially the name we choose because of the "gui" part.
Anyway we are stuck with this name for the time being... :scream:
